lib/bap/bap_project.ml

```diff
@@ -449,7 +458,7 @@ let compute_units ?package ?state main libs =
         Term.enum sub_t prog |> Seq.fold ~init:p ~f:(Term.append sub_t),
         {unit; info} :: l) in
   set_package package >>| fun () ->
-  main, libs, prog
+  main, List.rev libs, prog
```

Comment:
Not sure why it is there, you already reversed it in the dedup function.

Reply:
The libraries actually are accumulated in reverse order when we process them one-by-one, so to preserve the original order we reverse again. Does this make sense?